util/boac_utils.rb

```diff
@@ -295,6 +322,14 @@ def self.get_test_alert
   def self.get_student_notes(student)
     query = "SELECT * FROM notes WHERE sid = '#{student.sis_id}';"
     results = Utils.query_pg_db(boac_db_credentials, query)
+    result_ids = results.map { |r| r['id'] }
+
+    attach_query = "SELECT * FROM note_attachments WHERE note_id IN (#{result_ids.join(',')});"
+    attach_results = Utils.query_pg_db(boac_db_credentials, attach_query)
+
+    topic_query = "SELECT note_id, topic FROM note_topics WHERE note_id IN (#{result_ids.join(',')});"
+    topic_results = Utils.query_pg_db(boac_db_credentials, topic_query)
+
```

Comment:
Speeds up execution time for students with many notes using fewer, larger queries.